discovery.md

```diff
@@ -650,30 +630,49 @@ upon return of the `202 Accepted` from the original request.
-Discovery Endpoint. The Body of the request message MUST contain an array of
-zero or more Service Entries. For convenience to clients, each Service in the
-array MAY include the `id` or `epoch` attributes but they MUST be ignored by
-the Discovery Endpoint. Both of these attributes' values MUST be defined by
-the Discovery Endpoint.
+This MUST update the collection of Services in the Discovery Endpoint to
+contain all Services in the request or fail.  Any existing Service identified
+in the request will be completely replaced by the Service definition in the
+request.  The body of the request message MUST contain an array of zero or more
+Service entries.
+
+The `id` and `epoch` attributes of the Services MAY be omited. In such cases
+the discovery endpoint MUST assign an appropriate value to the omitted
+field(s).  An appropriate `id` MUST be globally unique.  An appropriate `epoch`
+MUST be greater than the any existing `epoch` for a service with the same `id`.
+Whether an `id` is given or supplied, any existing Service definition with that
+`id` MUST be replaced by the Service definition in the request and any assigned
+values. If an `epoch` is given in the request it MUST be greater than the
+`epoch` of any existing Service of the same `id` or the request MUST fail.
+
+If the Discovery Endpoint is unable to successully add every Service in the
+incoming request then an error MUST be generated and none of the specified
+Services are to be added or updated in the Discovery Endpoint.
+
+The Services in the request MUST be processed in the order in which they are
+given. While any error during the update MUST cause the rollback of any effect
+of the request, the processing of each Service in the request MUST be
+applied as if all previous Services in the request were already completed.
+This means that attribute uniqueness checking, such as on the `name`
+attribute, MUST be done on the new state of all Services as if all previously
+specified Services in the incoming request were already successfully processed.
 
-If any Service specified in the request matches the `name` of any existing
-Service, or the same `name` is used more than once within the request, then
-an error MUST be generated.
+For example, if a Discovery Endpoint has a Service called `MyService`, it is
+possible to rename that Service to `YourService` and then to add, or update,
+another Service to use `MyService` as its `name`. But this is only true if
+the rename of the first Service happens first the request message.
 
-If the Discovery Endpoint is unable to successully add all of the Services
-in the incoming request then an error MUST be generated and none of the
-specified Services are to be added to the Discovery Endpoint.
+Likewise, while it might be less than optimal, it is technically possible for
+a request to update a Service multiple times if the same `id` is used more
+than once within a request.
```

Comment:
this condenses the version with `?import` into this single endpoint definition to eliminate the multifaced behavior.